src/ints/bios_disk.cpp

```diff
@@ -99,34 +100,33 @@ void incrementFDD(void) {
 	CMOS_SetRegister(0x14, (Bit8u)(equipment&0xff));
 }
 
-void swapInDisks(void) {
-	bool allNull = true;
-	Bit32s diskcount = 0;
-	Bit32s swapPos = swapPosition;
-	Bit32s i;
-
-	/* Check to make sure that  there is at least one setup image */
-	for(i=0;i<MAX_SWAPPABLE_DISKS;i++) {
-		if(diskSwap[i]) {
-			allNull = false;
+template<typename T, size_t N>
+static size_t disk_array_prefix_size(const std::array<T, N> &images) {
+	size_t num = 0;
+	for (const auto &disk : images) {
+		if (!disk)
 			break;
-		}
+		num++;
 	}
+	return num;
+}
 
-	/* No disks setup... fail */
-	if (allNull) return;
+void swapInDisks(void) {
```

Comment:
A comment about what this function's purpose is would be helpful.
The function name doesn't say much, and the implementation manipulates global variables and then exits w/ `void`.
The only partial help are the two LOG_MSG's, and how you've nicely unrolled the while loops (which reads much better now, given we only ever deal with drive A and B).

Reply:
Yeah, I got confused and wasted some time twice already because I presumed this function is used for swapping CD images as well... In later commit I added a parameter to avoid this global extern variable, but I think I'll add documentation in `bios_disk.h` as well. Name of the function is not bad, but presumes that reader already knows how `boot` command swaps floppies.

Reply:
Awesome; thank you! 
(I'll be ready to read it and learn about them).
It's too bad there is almost no documentation in the code that says what's trying to accomplish (and the state going in versus state coming out); so refactoring is much harder.  Then again, it's better than incorrect documentation; so I suppose that's a bonus there's nothing to mislead us either.

Reply:
I added documentation (only to this one function and only to the header file). Included in the latest version of the review :)

Reply:
Read it; very helpful!